### ISSUE TITLE
faking menu items doesn't work

### ISSUE THREAD

**Issue by jhnc** [misd-service-development/drupal-cambridge-theme#41]
If I add everything into the menu system, navigation works fine. However page loading becomes unacceptably slow.

The main way I have found to avoid adding everything into the menu system appears to be to call menu_set_active_item() or menu_tree_set_path(). Unfortunately neither of these works with the theme.

menu_tree_set_path() appears to add entities to the correct part of the menu tree
but the vertical navigation is not expanded and breadcrumbs are broken.

menu_set_active_item() expands the navigation (mostly) correctly, but breadcrumbs lose the final crumb (eg. page d; desired trail = a->b->c->d ; calling menu_set_active_item(c) gives trail = a->b->d) causing operations such as editing to potentially operate on the wrong item.

I failed to attach three images that I hope illustrate the problem, so they are available at:

http://www.newton.ac.uk/tmp/jc254/pjl/1__menu_tree_set_path.png
http://www.newton.ac.uk/tmp/jc254/pjl/2__menu_set_active_item.png
http://www.newton.ac.uk/tmp/jc254/pjl/3__desired.png

**Comment by mrginglymus** [comment]
For those returning to this issue, the module [Menu Position](https://www.drupal.org/project/menu_position) might provide a quick solution.

**Comment by jhnc** [comment]
Yes, I would hope it to be possible to use menu_position for this but as a novice I couldn't work out how.

My recollection of my attempt to use it is that it would only have been helpful in my case when adding leaves to the menu because I couldn't work out how to nest placeholders without breaking breadcrumbs, active trail, or other parts of the projectlight navigation:

  * node1
    * [menuposition1]
      * node2
        * [menuposition2]
          * node3
            * [menuposition3]
              * node4
              * [...]
          * [...]
      * [...]

(In my case, mp1/2/3 might correspond to lists of programmes/workshops/seminars.)

Also, I think I couldn't work out how to dynamically add them into menus.
